model/QueueBroker/NewSqlQueueBroker.php

```diff
@@ -0,0 +1,271 @@
+            . ', '
+            . \common_Utils::toHumanReadablePhpString($this->getNumberOfTasksToReceive())
+            . ')';
+    }
+
+    protected function getPersistence(): ?common_persistence_SqlPersistence
+    {
+        if (is_null($this->persistence)) {
+            $this->persistence = $this->getServiceLocator()
+                ->get(common_persistence_Manager::SERVICE_ID)
+                ->getPersistenceById($this->persistenceId);
+        }
+
+        return $this->persistence;
+    }
+
+    private function getTableName(): string
+    {
+        return strtolower($this->getQueueNameWithPrefix());
+    }
+
+    /**
+     * Note: this method can be run multiple times because only the migrate queries (result of getMigrateSchemaSql) will be run.
+     *
+     * @inheritdoc
+     */
+    public function createQueue()
+    {
+        $persistence = $this->getPersistence();
+        /** @var AbstractSchemaManager $schemaManager */
+        $schemaManager = $persistence->getDriver()->getSchemaManager();
+
+        /** @var Schema $schema */
+        $schema = $schemaManager->createSchema();
+        $fromSchema = clone $schema;
+
+        if (in_array($this->getTableName(), $schemaManager->getTables())) {
+            $schema->dropTable($this->getTableName());
+        }
+        $queries = $persistence->getPlatForm()->getMigrateSchemaSql($fromSchema, $schema);
+
+        foreach ($queries as $query) {
+            $persistence->exec($query);
+        }
+
+        try {
+            $this->getSchemaProvider($schema)
+                ->setQueueName($this->getQueueName())
+                ->getSchema($schema, $this->getTableName());
+        } catch (Throwable $e) {
+            $this->logDebug('Schema of ' . $this->getTableName() . ' table already up to date.');
+        }
+
+        $queries = $persistence->getPlatForm()->getMigrateSchemaSql($fromSchema, $schema);
+
+        foreach ($queries as $query) {
+            $persistence->exec($query);
+        }
+
+        if ($queries) {
+            $this->logDebug('Queue ' . $this->getTableName() . ' created/updated in RDS.');
+        }
+    }
+
+    public function getSchemaProvider(Schema $schema): NewSqlSchema
+    {
+        return $this->getServiceLocator()->get(NewSqlSchema::class);
+    }
+
+    /**
+     * Insert a new task into the queue table.
+     *
+     * @param TaskInterface $task
+     * @return bool
+     */
+    public function push(TaskInterface $task)
+    {
+        return (bool)$this->getPersistence()->insert($this->getTableName(), [
+            'id' => $this->getUniquePrimaryKey(),
+            'message' => $this->serializeTask($task),
+            'created_at' => $this->getPersistence()->getPlatForm()->getNowExpression(),
+            'visible' => true,
+        ]);
+    }
+
+    /**
+     * Does the DBAL specific pop mechanism.
+     */
+    protected function doPop()
+    {
+        $this->getPersistence()->getPlatform()->beginTransaction();
```

Comment:
This method and `createQueue` method should be split in multiple private methods with proper names.

Here, for example I see 3 methods in one:

- `getMessagesToProcess()` (until the first if)
-  `makeMessagesInvisible()`
-  `pushPreFetchedMessages()`

All those can be called inside the try/catch